### PR TITLE
fix bgp address family to support multiple neighbours

### PR DESCRIPTION
**Pull request by KB-perByte**
##### SUMMARY

closing #422 #392 

- fix BGP address family to support multiple neighbors
- Inconsistent naming of class in the module fixed as per cli_rm_builders rule

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

bgp_address_family

### CODE REVIEW

File: plugins/module_utils/network/ios/config/bgp_address_family/bgp_address_family.py [hunk 1]
```diff
@@ -28,11 +28,11 @@
     Facts,
 )
 from ansible_collections.cisco.ios.plugins.module_utils.network.ios.rm_templates.bgp_address_family import (
-    Bgp_AddressFamilyTemplate,
+    Bgp_address_familyTemplate,
```
Comment (NilashishC) [inline plugins/module_utils/network/ios/config/bgp_address_family/bgp_address_family.py:31]: Not sure why we need to change these. These are generated by the cli_rm_builder `{{ resource|capitalize }}`.

Comment (KB-perByte) [reply plugins/module_utils/network/ios/config/bgp_address_family/bgp_address_family.py:31]: The module name is produced consistent with cli_rm_builder's naming pattern. What it occurs to me during scaffolding the RM builder was not used properly then. Addressed all the Class name changes in the PR.

File: plugins/module_utils/network/ios/argspec/bgp_address_family/bgp_address_family.py [hunk 2]
```diff
@@ -411,17 +413,17 @@ class Bgp_AddressFamilyArgs(object):
...
+                                    "type": "dict",
                                     "options": {
                                         "name": {"type": "str"},
                                         "in": {"type": "bool"},
                                         "out": {"type": "bool"},
                                     },
                                 },
-                                "route_map": {
-                                    "type": "dict",
+                                "route_maps": {
```
Comment (NilashishC) [inline plugins/module_utils/network/ios/argspec/bgp_address_family/bgp_address_family.py:424]: This is a breaking change.

File: docs/cisco.ios.ios_bgp_address_family_module.rst [hunk 3]
```diff
@@ -6681,7 +6681,7 @@ Examples
                     - name: test-route-in
                       in: true
                   route_server_client: true
-              network:
+              networks:
```
Comment (NilashishC) [inline docs/cisco.ios.ios_bgp_address_family_module.rst:6684]: Please remove this from docs for now.

File: tests/unit/modules/network/ios/test_ios_bgp_address_family.py [hunk 4]
```diff
@@ -750,3 +750,108 @@ def test_ios_bgp_address_family_parsed(self):
             "as_number": "65000",
         }
         self.assertEqual(parsed_list, result["parsed"])
+
+    def test_ios_bgp_address_family_merged_multiple_neighbor(self):
```
Comment (NilashishC) [inline tests/unit/modules/network/ios/test_ios_bgp_address_family.py:754]: These test cases do not assert that all the neighbors/networks are being read. It reads the input (from task) and pushes the commands, which I believe was working before too. You'd need a `gathered` or `parsed` test case with multiple neigbors/networks, and/or an idempotent version of the test cases you added.

**Review by NilashishC — APPROVED** [review]

**Pull request merged**